.github/workflows/api-binary-compatibility.yml

```diff
@@ -46,7 +46,7 @@ jobs:
           path: ~/.gradle/caches
           key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
           restore-keys: ${{ runner.os }}-gradle
-      - run: ./gradlew :iceberg-api:revapi
+      - run: ./gradlew clean :iceberg-api:revapi -x test -x integrationTest -x javadoc
```

Comment:
No, not entirely.

The problem, as outlined in the new PR https://github.com/apache/iceberg/pull/4989, was that the `actions/checkout` github action uses a depth of `1` by default.

So the "head" (master) branch is in a detached state, and thus the tags needed for the action to work (basically the output of `git describe`) was failing.

I _did_ remove the cache, which then caused the tests to at least run `revapi`. But they still passed because the plugin couldn't properly determine the old version (via the tag) and then the new version via the override in the `.palantir/revapi.yaml` file).

Let's continue the discussion in  https://github.com/apache/iceberg/pull/4989